protocol/communities/community.go

```diff
@@ -1320,20 +1321,64 @@ func (o *Community) Categories() map[string]*protobuf.CommunityCategory {
 	return response
 }
 
-func (o *Community) TokenPermissions() map[string]*protobuf.CommunityTokenPermission {
-	return o.config.CommunityDescription.TokenPermissions
+func (o *Community) tokenPermissions() map[string]*CommunityTokenPermission {
+	result := make(map[string]*CommunityTokenPermission, len(o.config.CommunityDescription.TokenPermissions))
+	for _, tokenPermission := range o.config.CommunityDescription.TokenPermissions {
+		result[tokenPermission.Id] = NewCommunityTokenPermission(tokenPermission)
+	}
+
+	// Non-privileged members should not see pending permissions
+	if o.config.EventsData == nil || !o.IsPrivilegedMember(o.MemberIdentity()) {
```

Comment:
`!o.IsPrivilegedMember(o.MemberIdentity())` this check shouldn't be needed

TODO: reject token permission events for ordinary members